File: buffer/gpu/src/jvmMain/java/com/wsr/gpu/elementwise/operation/JPlus.java
package com.wsr.gpu.elementwise.operation;

public class JPlus {
    public native void plusD0ToD1(float x, long y, long result, long runtime);

    public native void plusD1ToD0(long x, float y, long result, long runtime);

    public native void plusD1ToD1(long x, long y, long result, long runtime);

    public native void plusD1ToD2(long x, long y, int yi, int yj, int axis, long result, long runtime);

    public native void plusD1ToD3(long x, long y, int yi, int yj, int yk, int axis, long result, long runtime);

    public native void plusD2ToD1(long x, int xi, int xj, long y, int axis, long result, long runtime);

    public native void plusD2ToD3(
            long x,
            int xi,
            int xj,
            long y,
            int yi,
            int yj,
            int yk,
            int axis1,
            int axis2,
            long result,
            long runtime
    );

    public native void plusD3ToD1(long x, int xi, int xj, int xk, long y, int axis, long result, long runtime);

    public native void plusD3ToD2(
            long x,
            int xi,
            int xj,
            int xk,
            long y,
            int yi,
            int yj,
            int axis1,
            int axis2,
            long result,
            long runtime
    );

    public native void plusD3ToD4(
            long x,
            int xi,
            int xj,
            int xk,
            long y,
            int yi,
            int yj,
            int yk,
            int yl,
            int axis1,
            int axis2,
            int axis3,
            long result,
            long runtime
    );

    public native void plusD4ToD1(long x, int xi, int xj, int xk, int xl, long y, int axis, long result, long runtime);

    public native void plusD4ToD2(
            long x,
            int xi,
            int xj,
            int xk,
            int xl,
            long y,
            int yi,
            int yj,
            int axis1,
            int axis2,
            long result,
            long runtime
    );

    public native void plusD4ToD3(
            long x,
            int xi,
            int xj,
            int xk,
            int xl,
            long y,
            int yi,
            int yj,
            int yk,
            int axis1,
            int axis2,
            int axis3,
            long result,
            long runtime
    );
}

File: buffer/cpu/src/jvmAndroidMain/kotlin/com/wsr/knist/cpu/elementwise/operation/plus/JPlus.kt
package com.wsr.knist.cpu.elementwise.operation.plus


object JPlus {
    external fun plusD0ToD1(x: Float, y: Long, result: Long)

    external fun plusD1ToD0(x: Long, y: Float, result: Long)
    external fun plusD1ToD1(x: Long, y: Long, result: Long)
    external fun plusD1ToD2(x: Long, y: Long, yi: Int, yj: Int, axis: Int, result: Long)
    external fun plusD1ToD3(x: Long, y: Long, yi: Int, yj: Int, yk: Int, axis: Int, result: Long)

    external fun plusD2ToD1(x: Long, xi: Int, xj: Int, y: Long, axis: Int, result: Long)
    external fun plusD2ToD3(
        x: Long,
        xi: Int,
        xj: Int,
        y: Long,
        yi: Int,
        yj: Int,
        yk: Int,
        axis1: Int,
        axis2: Int,
        result: Long,
    )

    external fun plusD3ToD1(x: Long, xi: Int, xj: Int, xk: Int, y: Long, axis: Int, result: Long)
    external fun plusD3ToD2(
        x: Long,
        xi: Int,
        xj: Int,
        xk: Int,
        y: Long,
        yi: Int,
        yj: Int,
        axis1: Int,
        axis2: Int,
        result: Long,
    )
    external fun plusD3ToD4(
        x: Long,
        xi: Int,
        xj: Int,
        xk: Int,
        y: Long,
        yi: Int,
        yj: Int,
        yk: Int,
        yl: Int,
        axis1: Int,
        axis2: Int,
        axis3: Int,
        result: Long,
    )

    external fun plusD4ToD1(
        x: Long,
        xi: Int,
        xj: Int,
        xk: Int,
        xl: Int,
        y: Long,
        axis: Int,
        result: Long,
    )
    external fun plusD4ToD2(
        x: Long,
        xi: Int,
        xj: Int,
        xk: Int,
        xl: Int,
        y: Long,
        yi: Int,
        yj: Int,
        axis1: Int,
        axis2: Int,
        result: Long,
    )
    external fun plusD4ToD3(
        x: Long,
        xi: Int,
        xj: Int,
        xk: Int,
        xl: Int,
        y: Long,
        yi: Int,
        yj: Int,
        yk: Int,
        axis1: Int,
        axis2: Int,
        axis3: Int,
        result: Long,
    )
}
